routes/external/steam.js

```diff
@@ -58,21 +58,107 @@ router.get('/profile/:steamId', (req, res) => {
+      responseData = response.data.response.games;
+      for (let i = 0; i < responseData.length; i++){
+        let gameImageUrl = `http://media.steampowered.com/steamcommunity/public/images/apps/${responseData[i].appid}/${responseData[i].img_logo_url}.jpg`
+        responseData[i].image_url = gameImageUrl;
       }
-      return res.send(response.data.response.games)
     })
-    .catch(error => console.log(error))
+    .then(() => {
+        let promiseArray = [];
+        for (let i = 0; i < responseData.length; i++) {
+            const sani_name = responseData[i].name.replace(/[^0-9a-z'\s]/gi, ' ')
+                .replace(/[^0-9a-z\s]/gi, '')
+                .replace(/\s\s+/g, ' ')
+                .toLowerCase().split(" ").join("-");
+            const x = axios.get('https://api-v3.igdb.com/games', {
+                headers: {
+                    'Accept': 'application/json',
+                    'user-key': keys.igdbKey
+                },
+                data: `fields id; where slug = "${sani_name}";`
+            })
+            .then(response => {
+                const game = response.data[0];
+                if (game) {
+                    responseData[i].igdbId = game.id;
+                } else {
+                    responseData[i].igbdId = null;
+                }
+            })
+            promiseArray.push(x);
+        }
+        return Promise.all(promiseArray);
+    })
+    .then(() => { 
+        let promiseArray = [];
+        for (let i = 0; i < responseData.length; i++) {
+            // axios call to get achievements
+            const p1 = axios.get(`http://api.steampowered.com/ISteamUserStats/GetPlayerAchievements/v0001`, { params: {
+                key: keys.steamAPIKey,
+                appid: responseData[i].appid,
+                steamId
+            }})            
+            .then(response => {
+                // console.log(response.data.playerstats.achievements);
+                let achievementsAll = response.data.playerstats.achievements;
+                let achievementsCompleted = achievementsAll.filter(el => el.achieved === 1);
+                responseData[i].totalAchievements = achievementsAll.length;
+                responseData[i].completedAchievements = achievementsCompleted.length;
+            })
+            .catch(err => {
+                console.log("Error");
+                responseData[i].totalAchievements = "error";
+                responseData[i].completedAchievements = "error";
+            });
+            promiseArray.push(p1);
+            
+            if (responseData[i].igdbId) {
+                const p2 = axios.get('https://api-v3.igdb.com/time_to_beats', {headers: {
+                        'Accept': 'application/json',
+                        'user-key': keys.igdbKey
+                    }, 
+                    data: `fields *; where game = ${responseData[i].igdbId}; limit 50;`
+                })
+                .then(response => {
```

Comment:
Refactor, maybe pull out some methods (helpers). Very hard to read what's happening